sdk/iot/hub/tests/cmocka/test_az_iot_hub_client.c

```diff
@@ -60,8 +60,8 @@ static void test_az_iot_hub_client_init_succeed(void** state)
   assert_int_equal(
       az_iot_hub_client_init(&client, test_device_hostname, test_device_id, NULL), AZ_OK);
 
-  assert_string_equal(TEST_DEVICE_ID_STR, az_span_ptr(client._internal.device_id));
-  assert_string_equal(TEST_DEVICE_HOSTNAME_STR, az_span_ptr(client._internal.iot_hub_hostname));
+  assert_memory_equal(TEST_DEVICE_ID_STR, az_span_ptr(client._internal.device_id), _az_COUNTOF(TEST_DEVICE_ID_STR) - 1);
```

Comment:
I think you should check for the length of the span, not the length of the string - 1.
You are also missing a check for the actual len which should be the one you've written here.

To make the test even better, there should be a memset on all test buffers with anything _but_ '\0' to ensure things work as expected and we' don't have any NULLs added during the copy. The test should assert that the last char in the span is not null.

Reply:
Yes this will be addressed in another PR